palantir-java-format-api/src/main/java/com/palantir/javaformat/java/FormatterService.java

```diff
@@ -18,6 +18,19 @@
      * @return a list of {@link Replacement}s, sorted from low index to high index, without overlaps
      * @throws FormatterException if the input string cannot be parsed
      */
-    ImmutableList<Replacement> getFormatReplacements(
-            JavaFormatterOptions options, String input, Collection<Range<Integer>> ranges) throws FormatterException;
+    ImmutableList<Replacement> getFormatReplacements(String input, Collection<Range<Integer>> ranges)
+            throws FormatterException;
+
+    /**
+     * Formats an input string (a Java compilation unit), reflows strings and fixes imports.
+     *
+     * <p>Fixing imports includes ordering, spacing, and removal of unused import statements.
+     *
+     * @param input the input string
+     * @return the output string
+     * @throws FormatterException if the input string cannot be parsed
+     * @see <a href="https://google.github.io/styleguide/javaguide.html#s3.3.3-import-ordering-and-spacing">Google Java
+     *     Style Guide - 3.3.3 Import ordering and spacing</a>
+     */
+    String formatSourceReflowStringsAndFixImports(String input) throws FormatterException;
 }
```

Comment:
could we also rename this project from -api to -spi to emphasise it's a service provider interface??